library/src/main/java/com/timehop/stickyheadersrecyclerview/StickyRecyclerHeadersDecoration.java

```diff
@@ -88,27 +88,25 @@ public void onDrawOver(Canvas canvas, RecyclerView parent, RecyclerView.State st
       return;
     }
 
-    for (int i = 0; i < parent.getChildCount(); i++) {
-      View itemView = parent.getChildAt(i);
-      int position = parent.getChildPosition(itemView);
-      if (hasStickyHeader(i, position) || mHeaderPositionCalculator.hasNewHeader(position)) {
-        View header = mHeaderProvider.getHeader(parent, position);
-        Rect headerOffset = mHeaderPositionCalculator.getHeaderBounds(parent, header,
-            itemView, hasStickyHeader(i, position));
-        mRenderer.drawHeader(parent, canvas, header, headerOffset);
-        mHeaderRects.put(position, headerOffset);
+    RecyclerView.LayoutManager layoutManager = parent.getLayoutManager();
+    if ( layoutManager instanceof LinearLayoutManager ) {
```

Comment:
The LayoutManager will always be a LinearLayoutManager so you can just cast it.  I'd rather crash if somehow a non-linearlayout manager is attached than suppress the bug.